File: eCommerce/eCommerce.Services/Database/Review.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eCommerce.Services.Database
{
    public class Review
    {
        [Key]
        public int Id { get; set; }
        
        public int ReservationId { get; set; }
        
        [ForeignKey("ReservationId")]
        public Reservation Reservation { get; set; } = null!;
        
        public int UserId { get; set; }
        
        [ForeignKey("UserId")]
        public User User { get; set; } = null!;
        
        public int RestaurantId { get; set; }
        
        [ForeignKey("RestaurantId")]
        public Restaurant Restaurant { get; set; } = null!;
        
        public int Rating { get; set; }
        
        [MaxLength(1000)]
        public string? Comment { get; set; }
        
        public int? FoodQuality { get; set; }
        
        public int? ServiceQuality { get; set; }
        
        public int? AmbienceRating { get; set; }
        
        public int? ValueForMoney { get; set; }
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        public bool IsVerified { get; set; } = false;
    }
}

File: eCommerce/UI/ecommerce_mobile/lib/model/review.dart
class Review {
  final int id;
  final int reservationId;
  final int userId;
  final String userName;
  final int restaurantId;
  final String restaurantName;
  final int rating;
  final String? comment;
  final int? foodQuality;
  final int? serviceQuality;
  final int? ambienceRating;
  final int? valueForMoney;
  final DateTime createdAt;
  final bool isVerified;

  Review({
    required this.id,
    required this.reservationId,
    required this.userId,
    required this.userName,
    required this.restaurantId,
    required this.restaurantName,
    required this.rating,
    this.comment,
    this.foodQuality,
    this.serviceQuality,
    this.ambienceRating,
    this.valueForMoney,
    required this.createdAt,
    required this.isVerified,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id'] ?? 0,
      reservationId: json['reservationId'] ?? 0,
      userId: json['userId'] ?? 0,
      userName: json['userName'] ?? '',
      restaurantId: json['restaurantId'] ?? 0,
      restaurantName: json['restaurantName'] ?? '',
      rating: json['rating'] ?? 0,
      comment: json['comment'],
      foodQuality: json['foodQuality'],
      serviceQuality: json['serviceQuality'],
      ambienceRating: json['ambienceRating'],
      valueForMoney: json['valueForMoney'],
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : DateTime.now(),
      isVerified: json['isVerified'] ?? false,
    );
  }
}
